src/app/clients/edit/[id]/page.tsx

```diff
@@ -1,23 +1,40 @@
-import { Metadata } from "next";
+"use client";
 
-import React from "react";
-import { getSupabaseServerComponentClient } from "@/supabaseServer";
+import React, { useEffect, useState } from "react";
+import supabase from "@/supabaseClient";
 import ClientForm from "@/app/clients/form/ClientForm";
 import { Errors, FormErrors } from "@/components/Form/formFunctions";
 import autofill from "@/app/clients/edit/[id]/autofill";
 import { fetchClient, fetchFamily } from "@/common/fetch";
+import { Schema } from "@/databaseUtils";
 
 interface EditClientsParameters {
     params: { id: string };
 }
 
-const EditClients: ({ params }: EditClientsParameters) => Promise<React.ReactElement> = async ({
+const EditClients: ({ params }: EditClientsParameters) => React.ReactElement = ({
     params,
 }: EditClientsParameters) => {
-    const supabase = getSupabaseServerComponentClient();
-    const clientData = await fetchClient(params.id, supabase);
-    const familyData = await fetchFamily(clientData.family_id, supabase);
-    const initialFields = autofill(clientData, familyData);
+    const [clientData, setClientData] = useState<Schema["clients"]>();
+    const [familyData, setFamilyData] = useState<Schema["families"][]>();
+
+    useEffect(() => {
+        (async () => {
+            if (params.id) {
+                setClientData(await fetchClient(params.id, supabase));
+            }
+        })();
+    }, [params.id]);
+    useEffect(() => {
+        (async () => {
+            if (clientData?.family_id) {
+                setFamilyData(await fetchFamily(clientData.family_id, supabase));
+            }
+        })();
```

Comment:
should we log an error when fetchClient or fetchFamily fails? In which case it might be easier to read with `.then` and `.catch` so we can get rid of the `(async () => { ... })()` structure

i.e.
```ts
if (params.id) {
  fetchClient(params.id, supabase)
    .then(client => setClientData(client))
    .catch(error => {
      logError(...)
      // maybe even set an error message to display on the page, rather than showing an empty page
    })
}
```

Reply:
Cool, have done